### PR TITLE
Make icons publishable to NPM

### LINKED ISSUE #152: Consider publishing to NPM
Would you consider publishing icons as an NPM package?

It could make it easier to download and version.

### PR DESCRIPTION
**Pull request by metonym**
Closes #152

This PR makes the icons publishable to the NPM registry.

At a minimum, it does the following:

- Add `package.json#files` to specify the folder to include to publish (`public/icons`)
- Update the `README` to include NPM as a download source

To test this locally, you can run `yarn pack` to simulate how the library will be packaged when publishing.

Unzipping the tarball should produce the following structure:

```
LICENSE
package.json
public/icons/*
README.md
```

## Recommendations

1. I would *highly* recommend a [CHANGELOG.md](https://keepachangelog.com/en/1.1.0/) be created to document changes to icons for subsequent publishes. NPM uses [semantic versioning](https://semver.org/).
2. I would recommend including some `package.json#keywords` as additional metadata.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,11 +14,16 @@ The project is hosted by the public health not-for-profit [Resolve to Save Lives
...
-Icons are available in several formats and in a few ways:
+Icons are available in several formats:
 
 - All icons are produced in "outline", "filled", and "negative" styles
 - Each icon is available in SVG and in PNG (48px and 96px)
-- Icons can be downloaded individually via [our website](https://healthicons.org), downloaded together in [a ZIP file](https://healthicons.org/icons.zip), or accessed under the [public/icons](https://github.com/resolvetosavelives/healthicons/tree/main/public/icons) folder in this GitHub project.
+
+Icons can be downloaded individually in a few ways:
+
+- [From our website](https://healthicons.org)
```
Comment (sgarrity) [inline README.md:24]: I would say "Individually from our website".

File: README.md [hunk 2]
```diff
@@ -14,11 +14,16 @@ The project is hosted by the public health not-for-profit [Resolve to Save Lives
...
+- As [a ZIP file](https://healthicons.org/icons.zip)
```
Comment (sgarrity) [inline README.md:25]: "Together as a ZIP file"

File: README.md (hunk 1 above)
Comment (metonym) [reply README.md]: Amended the commit

File: package.json [hunk 3]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "health-icons",
+  "name": "healthicons",
```
Comment (metonym) [inline package.json:2]: NPM uses `package.json#name` for [unscoped packages](https://docs.npmjs.com/creating-and-publishing-unscoped-public-packages).

**Comment by nburka** [comment]
This is on my TODO list to review.

**Comment by nburka** [comment]
@metonym sorry for the super slow response here. Life got really busy for me.

Just FYI - I'm testing this now. After some headaches with NPM/Yarn authentication, I've managed to publish a 1.0.0 version of the NPM package https://www.npmjs.com/package/healthicons

I'm going to then try to tie this into the CI pipeline.

**Comment by metonym** [comment]
Fantastic, thank you!

**Comment by nburka** [comment]
@metonym when I publish it - it includes all of the dependencies of the website. Is that just to be expected because the package.json contains all of the deps? Or is there some way to avoid that and just publish the icons by themselves? https://www.npmjs.com/package/healthicons?activeTab=dependencies

**Comment by metonym** [comment]
@nburka Nice catch – I agree that having the website dependencies in `package.json#dependencies` is not ideal.

I believe the [`package.json` file](https://docs.npmjs.com/cli/v9/configuring-npm/package-json) is required for publishing a package to NPM. However, within the file, only `name` and `version` fields are required.



> Like publishing a stripped down version of package.json ?

As you mentioned, I think this would be the preferred option.

1. Is it possible to simply move the `dependencies` to `devDependencies`?
2. Alternatively, a script could be written to create a temporary folder that copies over the icons, README, package.json, LICENSE, and then deletes the `dependencies` fields in `package.json`.

**Comment by nburka** [comment]
I've started rethinking this. Maybe we should split the repos. This came up in another github issue https://github.com/resolvetosavelives/healthicons/issues/195#issuecomment-1948671274

**Comment by nburka** [comment]
@metonym I'm really really sorry for letting this sit so long. I finally sat down to figure this issue out, and then realized that YOU totally did it already using the `files` option in the package.json. I didn't even see your comment about that or the change in the PR. That totally does what we need.

This PR was blocked because of a rebasing issue, but I just copied your changes into a fresh branch and am now getting ready to merge it. I've also released an initial NPM package with just the icons, no website, this time.

**Pull request closed without merging**